pkg/render/node.go

```diff
@@ -932,10 +942,9 @@ func (c *nodeComponent) nodeLivenessReadinessProbes() (*v1.Probe, *v1.Probe) {
 	return lp, rp
 }
 
-// nodeMetricsService creates a Service which exposes the calico/node metrics
-// reporting endpoint.
+// nodeMetricsService creates a Service which exposes the calico/node metrics reporting endpoint.
 func (c *nodeComponent) nodeMetricsService() *v1.Service {
-	return &v1.Service{
+	service := v1.Service{
```

Comment:
This code seems to always create a service for the nodeMetrics, but conditionally attach ports to it. Is that right? I think we want to delete the service if we're not using it.

Reply:
Ok

Reply:
@ozdanborne sadly, simply removing the service and returning `nil` here won't delete the service - we'd need to write special-case logic to delete it.

I had the same thought, but I believe what @rafaelvanoni has written here right now is a good pragmatic solution. If we simply returned nil here, then going from a port to no port would result in the service remaining unchanged.

Reply:
Ah neat. SGTM

Reply:
FWIW, this wouldn't be too hard. There's another PR that does this exact thing: https://github.com/tigera/operator/pull/382

Reply:
Ok